tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/util/BRunUtil.java

```diff
@@ -712,12 +714,24 @@ private static Object getJVMValue(BType type, BValue value) {
                     Object value = ((BLangLiteral) expr).value;
                     valueSpace.add(value);
                 }
-                return new org.ballerinalang.jvm.types.BFiniteType(null, valueSpace);
+                return new org.ballerinalang.jvm.types.BFiniteType(null, valueSpace,
+                        getMask(finiteType.isNullable(), finiteType.isAnydata(), finiteType.isPureType()));
             default:
                 throw new RuntimeException("Function argument for type '" + type + "' is not supported");
         }
     }
 
+    private static int getMask(boolean nilable, boolean isAnydata, boolean isPureType) {
+        int mask = nilable ? TypeFlags.NILABLE : 0;
+        if (isAnydata) {
+            mask = TypeFlags.addToMask(mask, TypeFlags.ANYDATA);
```

Comment:
a value being anydata implies that it is pure right? should we update this as 
```ballerina
        if (isAnydata) {
            mask = TypeFlags.addToMask(mask, TypeFlags.ANYDATA);
            mask = TypeFlags.addToMask(mask, TypeFlags.PURETYPE);
        } else if (isPureType) {
            mask = TypeFlags.addToMask(mask, TypeFlags.PURETYPE);
        }
```

Reply:
We could do this in multiple places, if there could potentially be a gain.

Reply:
I think its safe to depend on the respective flag to make it future proof. If someday we update the compiler type checking (in such a way that puretype and anydata doesn't have a direct relationship), then we don't have to worry about the runtime.